File: Learn/6. Queue/stackQueue.java
import java.util.Stack;

public class stackQueue {

    static class queue {
        private Stack<Integer> mainStack = new Stack<>();
        private Stack<Integer> secondStack = new Stack<>();

        public void enqueue(Integer data) {
            moveItemFromStack(mainStack, secondStack);
            mainStack.push(data);
            moveItemFromStack(secondStack, mainStack);
        }

        public boolean isEmpty() {
            return mainStack.isEmpty();
        }

        public Integer dequeue() {
            return mainStack.pop();
        }

        public Integer peek() {
            return mainStack.peek();
        }

        public void print() {
            System.out.println(mainStack);
        }

        private void moveItemFromStack(Stack<Integer> firstStack, Stack<Integer> secondStack) {
            while (firstStack.isEmpty() != true) {
                secondStack.push(firstStack.pop());
            }
        }

    }

    public static void main(String[] args) {
        queue test = new queue();
        test.enqueue(1);
        test.enqueue(2);
        test.enqueue(3);
        Integer frontValue = test.dequeue();
        Integer frontValuePeek = test.peek();
        System.out.println(frontValue);
        System.out.println(frontValuePeek);
        test.print();
    }

}

File: 6. Queue/priorityQueue/src/stackQueue.js
class Queue {
    constructor(){
      this.mainStack = [];
      this.secondStack = [];
    }

    enqueue = (data) =>{
        this.moveItemFromStack(this.mainStack, this.secondStack);
        this.mainStack.push(data);
        this.moveItemFromStack(this.secondStack, this.mainStack);
    }
    
    dequeue = () =>{
        if(this.isEmpty()){
            throw new Error("Queue is empty");
        }
        return this.mainStack.pop();
    }

    peek = () =>{
        if(this.isEmpty()){
            throw new Error("Queue is empty");
        }
        return this.mainStack[this.mainStack.length -1];
    }

    isEmpty = () =>{
        if(this.mainStack.length > 0 != true){
            return true;
        }else{
            return false;
        }
    }

    moveItemFromStack(firstStack, secondStack){
        while(firstStack.length > 0){
            secondStack.push(firstStack.pop());
        }
    }
}


module.exports = Queue;